packages/react-native-executorch/common/rnexecutorch/threads/HighPerformanceThreadPool.h

```diff
@@ -0,0 +1,367 @@
+// HighPerformanceThreadPool.h
+#pragma once
+
+#include <algorithm>
+#include <atomic>
+#include <chrono>
+#include <condition_variable>
+#include <executorch/extension/threadpool/cpuinfo_utils.h>
+#include <fstream>
+#include <functional>
+#include <future>
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <pthread.h>
+#include <queue>
+#include <rnexecutorch/Log.h>
+#include <sched.h>
+#include <sys/resource.h>
+#include <thread>
+#include <unistd.h>
+#include <vector>
+
+#ifdef __APPLE__
+#include <sys/syscall.h>
+#endif
+
+#ifdef __ANDROID__
+#include <sys/types.h>
+#endif
+
+namespace rnexecutorch {
+namespace threads {
+
+enum class Priority { LOW, NORMAL, HIGH, REALTIME };
+
+struct ThreadConfig {
+  bool pinToPerformanceCores{true};
+  std::string namePrefix{"RN_ET_Worker"};
+};
+
+class HighPerformanceThreadPool {
+public:
+private:
```

Comment:
Access specifiers are off in the `HighPerformanceThreadPool` class. Prefer `public` members at the very top, and then protected and private.